samples/storage_example.py

```diff
@@ -0,0 +1,88 @@
+"""Sample usage of the storage API.
+
+steps to run:
+    1. pip install .
+    2. docker pull postgres
+    3. docker run --name postgresDB -p 5455:5432 -e POSTGRES_USER=postgres \
+        -e POSTGRES_PASSWORD=postgres -e POSTGRES_DB=postgres -d postgres
+    4. python storage_example.py
+"""
+
+import dataclasses
+from typing import List
+from datetime import datetime
+from sqlalchemy.orm import Session
+from sqlalchemy.engine import Row
+
+import buildflow as flow
+
+
+@dataclasses.dataclass
+class Person:
+    # ID is a special type used to index the rows in the table. It
+    # autoincrements with every new row. It should be used when no PrimaryKey
+    # is provided.
+    id: flow.ID() = None
+    # PrimaryKey is a special type that is used to index the rows in the table.
+    name: flow.PrimaryKey(str) = ''
+    # All base python types are supported as columns
+    age: int = 0
+    hobbies: List[str] = dataclasses.field(default_factory=list)
+    birthday: datetime = dataclasses.field(default_factory=datetime)
+
+
+postgres = flow.Postgres(user='postgres',
+                         password='postgres',
+                         host='localhost:5455',
+                         database='postgres')
+
+
+@flow.storage(provider=postgres, schema=Person, table_name='persons')
```

Comment:
What if someone wanted to write to multiple tables inside one session? Can that just attach multiple decorators?

Something like:

```
@flow.storage(provider=postgres, schema=Person, table_name='persons')
@flow.storage(provider=postgres, schema=City, table_name='cities')
def write_data(p: Person, c: City, session: Session):
   session.add(p)
   session.add(c)
   session.commit()
```

I think this will probably be a requirement for transactional databases cause you don't want one write to succeed and the other fail, and then end up in a bad state.